presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -212,6 +213,23 @@ private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> s
                 DoubleStream::max);
     }
 
+    private Estimate calculateDataSize(Map<String, PartitionStatistics> statisticsByPartitionName, String column, Estimate rowCount)
+    {
+        return summarizePartitionStatistics(
+                statisticsByPartitionName.values(),
+                column,
+                hiveColumnStatistics -> {
+                    if (rowCount.isValueUnknown() || !hiveColumnStatistics.getAverageColumnLength().isPresent()) {
+                        return OptionalDouble.empty();
+                    }
+
+                    OptionalLong nullCount = hiveColumnStatistics.getNullsCount();
+                    double nonNulls = nullCount.isPresent() ? rowCount.getValue() - nullCount.getAsLong() : rowCount.getValue();
+                    return OptionalDouble.of(nonNulls * hiveColumnStatistics.getAverageColumnLength().getAsDouble());
+                },
+                DoubleStream::average);
```

Comment:
`DoubleStream::sum` ?
Even then, partitions without stats for given column (being skipped in `summarizePartitionStatistics`) or partitions without rowCount and/or AverageColumnLength (being skipped here), will not be part of the computed stat value. So, probably, `DoubleStream::average` and then `* number of all partitions` ?

Reply:
I would suggest taking the `sum`, and passing the total selected partition count. Than you can count the partition with the statistics and project the total number to the total number of partitions. We do that when estimating the `row_count` statistic.